src/main/java/wooteco/subway/application/LineService.java

```diff
@@ -108,6 +112,27 @@ public PathResponse findPath(final Long sourceId, final Long targetId) {
                 .orElseThrow(() -> new NoSuchStationException(targetId));
 
         Path path = subwayMap.calculatePath(sourceStation, targetStation);
-        return PathResponse.from(path.getStations(), path.getDistance(), FarePolicy.calculateFare(path.getDistance()));
+
+        List<Line> passingLines = mapLineIdToLine(path.getPassingLineIds());
+        int extraFare = calculateMostExpensiveExtraFare(passingLines);
+
+        return PathResponse.from(path.getStations(), path.getDistance(),
+                new FarePolicy(FareDiscountPolicyFactory.create(age)).calculateFare(path.getDistance(), extraFare));
+    }
+
+    private int calculateMostExpensiveExtraFare(final List<Line> passingLines) {
+        return passingLines.stream()
+                .map(Line::getExtraFare)
+                .max(Integer::compareTo)
+                .orElse(0);
+    }
+
+    private List<Line> mapLineIdToLine(final Set<Long> passingLineIds) {
```

Comment:
복수의 아이디를 복수의 노선으로 치환하기에 메서드명으로 표현해도 괜찮겠어요.

```suggestion
    private List<Line> mapLineIdsToLines(final Set<Long> passingLineIds) {
```